### ISSUE TITLE
Data refresh DAG reports error despite success of data refresh

### ISSUE THREAD

**Issue by stacimc** [WordPress/openverse-catalog#437]
## Description

The `audio_data_refresh` DAG successfully triggers an audio data refresh, but the `wait_for_completion` task fails with the exception:

```
 File "/usr/local/airflow/openverse_catalog/dags/data_refresh/dag_factory.py", line 93, in response_check_wait_for_completion
    raise AirflowException("Error triggering data refresh.")
```

Suggesting there was an error completing the task on the data refresh server. But the task appears to have completed successfully, and the data refresh server logs no errors.

This ticket should be used for investigation. If the issue is discovered to be on the data refresh server, we should open a new ticket in `openverse-api`.

## Reproduction

Note this only seems to be happening in production.

## Resolution

- [ ] 🙋 I would be interested in resolving this bug.

**Comment by stacimc** [comment]
The source of the problem is unfortunately the elasticsearch reindexing step (the final step of the data refresh). When running locally, we immediately and synchronously index the table before exiting the data refresh task. In staging/production, [we schedule multiple `indexer_worker` ](https://github.com/WordPress/openverse-api/blob/main/ingestion_server/ingestion_server/indexer.py#L414)instances to perform a distributed index. The data refresh task schedules the workers, then [logs and exits immediately](https://github.com/WordPress/openverse-api/blob/95820a73f090f2627ab516abe0a8b6bed363190d/ingestion_server/ingestion_server/tasks.py#L119).

This is really unfortunate because it means that the `status_check` endpoint can't really be used to check the status of the entire _data refresh_ outside of the local env. Not only will it report that the task is no longer `active` as soon as the work is scheduled, it also [misleadingly reports that there is an error](https://github.com/WordPress/openverse-api/blob/95820a73f090f2627ab516abe0a8b6bed363190d/ingestion_server/ingestion_server/api.py#L138) (because the task 'stopped' (is inactive) before `percent_completed == 100`).

Even worse, the elasticsearch reindexing step is very specifically the step that we need to make sure does not run concurrently. That means until this is resolved, we cannot turn on both data refresh DAGs (we can however run them "manually" by turning on one at a time 😢)

**Comment by stacimc** [comment]
The _ideal_ path forward would be to keep track of the worker information for the data refresh task, and only update the task status when all of the workers are done or when an error is detected. We'd also need to fix the error detection, which would require defining the error states.

There is an existing [WorkerFinishedResource](https://github.com/WordPress/openverse-api/blob/95820a73f090f2627ab516abe0a8b6bed363190d/ingestion_server/ingestion_server/api.py#L142) that I'm taking a look at.

**Comment by zackkrida** [comment]
Great analysis that explains the problem really clearly. I was so curious what the dev/prod discrepancy was, unfortunate it's something that will need such a comprehensive resolution.

**Comment by AetherUnbound** [comment]
Should be resolved in WordPress/openverse-api#592!
